packages/react-components/src/components/tooltip-xy-component.tsx

```diff
@@ -0,0 +1,123 @@
+
+    constructor(props: unknown) {
+        super(props);
+        this.state = {
+            tooltipData: undefined,
+            onDisplay: false,
+            inTooltip: false,
+        };
+        this.divRef = React.createRef();
+    }
+
+    render(): React.ReactNode {
+        let zeros = 0;
+        let allZeros = false;
+        if (this.state.tooltipData?.zeros && this.state.tooltipData.zeros > 0) {
+            zeros = this.state.tooltipData?.zeros;
+        }
+        if (this.state.tooltipData?.dataPoints?.length === 0) {
+            allZeros = true;
+        }
+
+        return (
+            <div
+                ref={this.divRef}
+                onMouseEnter={() => {
+                    this.setState({ inTooltip: true });
+                    if (this.timerId) {
+                        clearTimeout(this.timerId);
+                        this.timerId = undefined;
+                    }
+                }}
+                onMouseLeave={() => {
+                    this.setState({ tooltipData: { opacity: 0, transition: '0s' }, inTooltip: false });
+                }}
+                style={{
+                    padding: 10,
+                    position: 'fixed',
+                    textAlign: 'left',
+                    maxHeight: '250px',
+                    overflow: 'auto',
+                    color: 'white',
+                    border: '1px solid transparent',
+                    backgroundColor: 'rgba(51, 122, 183, 0.9)',
+                    borderRadius: 3,
+                    top: this.state.tooltipData?.top,
+                    bottom: this.state.tooltipData?.bottom,
+                    left: this.state.tooltipData?.left,
+                    opacity: this.state.tooltipData?.opacity ? this.state.tooltipData.opacity : 0,
+                    transition: this.state.tooltipData?.transition ? 'opacity ' + this.state.tooltipData.transition : 'opacity 0.3s',
+                    zIndex: 999
+                }}
+            >
+                <p style={{margin: '0 0 5px 0'}}>{this.state.tooltipData?.title}</p>
+                <ul style={{padding: '0'}}>
+                    {this.state.tooltipData?.dataPoints?.map((point: { color: string, background: string, label: string, value: string }, i: number) =>
+                        <li key={i} style={{listStyle: 'none', display: 'flex', marginBottom: 5}}>
+                            <div style={{
+                                height: '10px',
+                                width: '10px',
+                                margin: 'auto 0',
+                                border: 'solid thin',
+                                borderColor: point.color,
+                                backgroundColor: point.background
+                                }}
+                            ></div>
+                            <span style={{marginLeft: '5px'}}>{point.label} {point.value}</span>
+                        </li>
+                    )}
+                </ul>
```

Comment:
I'm not sure what is the cause but odd behavior: Initially, the tooltip can go beyond the bounds of the xy-output-component (for example, over the right-side Outline widget). But after some window manipulation (for example, collapsing the left-side Trace Viewer widget) then the tooltip can no longer go beyond the xy-output-component bounds, it gets clipped to where the Outline widget starts.

Reply:
Changed the "position" property of the tooltip from "fixed" to "absolute", which resolves the issue.